responsive-test-utils/src/main/java/dev/responsive/kafka/store/KVStoreStub.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2023 Responsive Computing, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.responsive.kafka.store;
+
+import dev.responsive.utils.Iterators;
+import java.time.Duration;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.TreeMap;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.common.utils.Time;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.state.KeyValueIterator;
+
+public class KVStoreStub {
+
+  private final NavigableMap<Bytes, byte[]> records = new TreeMap<>();
+  private final Map<Bytes, Long> expiryTimes = new LinkedHashMap<>();
```

Comment:
The class itself is not new, but there are some significant changes here. Mainly the difference is going from a single, simple key-value map to a navigable key-value map (to support `range` and `all`) and a separate ordered map with the expiration deadline for each key.

It's pretty much a quick version of Confluent's (in)famous interview question on window processing